File: LEDControl/Database/Models/ConvertVideo.cs
using System;
using System.ComponentModel.DataAnnotations;
using Xabe.FFmpeg;

namespace LEDControl.Database.Models;

public class ConvertVideo
{
    [Key]
    public Guid Id { get; set; }
    
    [MaxLength(250)]
    public string Link { get; set; }

    [MaxLength(250)]
    public string Title { get; set; }
    public int DownloadProgress { get; set; }
    public int ConvertProgress { get; set; }
    public ConvertStatus ConvertStatus { get; set; }
    public DateTime Created { get; set; }

    public ConversionPreset ConversionPreset { get; set; }
    public int Bitrate { get; set; } = 320_000;

    [MaxLength(250)]
    public string Hint { get; set; }
    
}

File: LEDControl-Client/src/app/models/convert-video.ts
export class ConvertVideo {
  id: string;
  link: string;
  title: string;
  hint: string;
  convertProgress: number;
  downloadProgress: number;
  convertStatus: ConvertStatus;
  conversionPreset: ConversionPreset;
}

export enum ConvertStatus{
  Waiting,
  Processing,
  Done,
  Failed
}

export enum ConversionPreset {
  VerySlow,
  Slower,
  Slow,
  Medium,
  Fast,
  Faster,
  VeryFast,
  SuperFast,
  UltraFast,
}
